odo/backends/sql.py

```diff
@@ -455,29 +465,28 @@ def append_anything_to_sql_Table(t, o, **kwargs):
 
 @append.register(sa.Table, sa.Table)
 def append_table_to_sql_Table(t, o, **kwargs):
-    # This condition is an ugly kludge and should be removed once
-    # https://github.com/dropbox/PyHive/issues/15 is resolved
-    if t.bind.name == o.bind.name == 'hive':
-        with t.bind.connect() as conn:
-            conn.execute('INSERT INTO TABLE %s SELECT * FROM %s' %
-                         (t.name, o.name))
-        return t
-
     s = sa.select([o])
     return append(t, s, **kwargs)
 
 
 @append.register(sa.Table, sa.sql.Select)
-def append_select_statement_to_sql_Table(t, o, **kwargs):
-    if not o.bind == t.bind:
-        return append(t, convert(Iterator, o, **kwargs), **kwargs)
-
-    assert o.bind.has_table(t.name, t.schema), \
+def append_select_statement_to_sql_Table(t, o, bind=None, **kwargs):
+    tbind = getbind(t, bind)
+    obind = getbind(o, bind)
+    if not tbind == obind:
+        return append(
+            t,
+            convert(Iterator, o, bind=bind, **kwargs),
+            bind=bind,
+            **kwargs
+        )
+
+    assert obind.has_table(t.name, t.schema), \
         'tables must come from the same database'
 
     query = t.insert().from_select(o.columns.keys(), o)
 
-    with o.bind.connect() as conn:
+    with obind.connect() as conn:
```

Comment:
maybe this should be `bind` just for clarity? i realize that if `bind is not None` then `obind == bind`, but IMO it reads better if you use `bind` here.